pyam/core.py

```diff
@@ -757,26 +758,42 @@ def rename(self, mapping=None, inplace=False, append=False,
         if not inplace:
             return ret
 
-    def convert_unit(self, conversion_mapping, inplace=False):
-        """Converts units based on provided unit conversion factors
+    def convert_unit(self, current, to=None, factor=None, inplace=False):
+        """Converts a unit using a given factor or the pint package
+
+        The `pint package <https://pint.readthedocs.io>`_ natively handles
+        conversion of standard (SI) units (e.g., exajoule to terawatt-hours,
+        :code:`EJ -> TWh`). It can also parse combined units (e.g.,
+        exajoule per year, :code:`EJ/yr`).
+
+        The :py:class:`pint.UnitRegistry` used by default loads additional
+        unit definitions relevant for integrated assessment models and energy
+        systems analysis from the `IIASA energy-units
+        <https://github.com/iiasa/energy-units>`_ repository.
+
+        You can access the :py:class:`pint.UnitRegistry` used by :class:`pyam`
+        via :func:`pint.get_application_registry`.
 
         Parameters
         ----------
-        conversion_mapping: dict
-            for each unit for which a conversion should be carried out,
-            provide current unit and target unit and conversion factor
-            {<current unit>: [<target unit>, <conversion factor>]}
+        current: str (or mapping, deprecated)
+            name of current unit (to be converted from)
+        to: str
+            name of new unit (to be converted to)
+        factor: value, optional
+            conversion factor if given, otherwise defaults to the application
+            :py:class:`pint.UnitRegistry`
         inplace: bool, default False
             if True, do operation inplace and return None
         """
-        ret = copy.deepcopy(self) if not inplace else self
-        for current_unit, (new_unit, factor) in conversion_mapping.items():
-            factor = pd.to_numeric(factor)
-            where = ret.data['unit'] == current_unit
-            ret.data.loc[where, 'value'] *= factor
-            ret.data.loc[where, 'unit'] = new_unit
-        if not inplace:
-            return ret
+        # TODO: deprecate using `dict` in next release (>=0.6.0)
+        # TODO: make `to` required
+        if isinstance(current, dict) and to is None and factor is None:
+            deprecation_warning('Use explicit keyword arguments instead!',
+                                'Using a dictionary to convert units')
```

Comment:
General comment - could you explicitly use the kwarg name (helps for future and current eyes to know what is being called), so, e.g., `deprecation_warning('..', type='..')`?